app/views/pets/show.html.erb

```diff
@@ -2,7 +2,11 @@
 
 <div class="pets">
 <section class="pets-<%= @pet.id %>">
-  <button><%= link_to 'Favorite', "/favorites/#{@pet.id}", method: :patch %></button>
+  <% if favorite.contents[@pet.id.to_s].nil? %>
+    <button><%= link_to 'Favorite', "/favorites/#{@pet.id}", method: :patch %></button>
+  <% else %>
+    <button><%= link_to 'Remove', "/favorites/#{@pet.id}", method: :delete %></button>
+  <% end %>
```

Comment:
Any logic involving different views will go in here and not in the controller.
For example, a page might look different depending on admin or regular customer so certain functions won't display. That logic will typically belong in views. Controllers are redirect logic, SAD paths, renders